src/sdk/pynni/nni/_graph_utils.py

```diff
@@ -0,0 +1,446 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+import logging
+import queue
+import re
+from collections import defaultdict
+import torch
+from torch.utils.tensorboard._pytorch_graph import NodePy, NodePyIO, NodePyOP, GraphPy
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats
+from tensorboard.compat.proto.versions_pb2 import VersionDef
```

Comment:
which package is this `tensorboard` from? tensorflow?

Reply:
Yes, from tensorflow, moved.